sunpy/map/sources/goes.py

```diff
@@ -0,0 +1,77 @@
+"""GOES SXI Map subclass definitions"""
+from __future__ import absolute_import, division, absolute_import
+#pylint: disable=W0221,W0222,E1101,E1121
+
+__author__ = "Steven Christe"
+__email__ = "steven.christe@nasa.gov"
+
+from astropy.visualization import LogStretch
+from astropy.visualization.mpl_normalize import ImageNormalize
+from matplotlib.colors import LogNorm
+from sunpy.map import GenericMap
+from sunpy.cm import cm
+from sunpy.map.sources.source_type import source_stretch
+from sunpy.coordinates import get_sunearth_distance, get_sun_B0
+from sunpy import sun
+import astropy.units as u
+
+__all__ = ['SXIMap']
+
+
+class SXIMap(GenericMap):
+    """GOES SXI Image Map
+
+    The Solar X-ray Imager (SXI) provides uninterrupted, full-disk, soft X-ray
+    solar images, with a 1 min cadence and a single-image (adjustable) dynamic
+    range near 100. A set of metallic thin-film filters provides temperature
+    discrimination in the 0.6–6.0 nm bandpass. The spatial resolution of
+    approximately 10 arcsec FWHM is sampled with 5 arcsec pixels.
+
+    SXI was first launched on 23 July 2001 on NOAA’s GOES-12 satellite and has
+    obtained uninterrupted images starting on 22 January 2003.
+
+    References
+    ----------
+    * `Mission/Instrument Page <https://sxi.ngdc.noaa.gov/index.html>`_
+    * `Data notes and Fits headers <https://sxi.ngdc.noaa.gov/sxi_data_notes.html>`_
+    * `Official SXI Data Book <https://sxi.ngdc.noaa.gov/images/section06.pdf>`_
+    * `Instrument Paper<https://sxi.ngdc.noaa.gov/images/2005_SxiInstrumentOperationsAndData.pdf>`_
+    * `SXI Performance Paper <https://sxi.ngdc.noaa.gov/images/2005_SxiPerformance.pdf>`__
```

Comment:
I would prefer provide DOI links for papers rather than links to noaa sites. Mostly for two reasons, doi links are less likely to change, and that will make easier to look how many papers we reference in the library.

Reply:
Good point. I'll fix.